ansible_base/lib/abstract_models/common.py

```diff
@@ -41,6 +41,46 @@ class ModifiableModel(models.Model):
     class Meta:
         abstract = True
 
+    trivial_fields = ["modified", "modified_by"]
+
+    def _has_non_trivial_changes(self, update_fields=None):
+        """Check if any non-timestamp fields have changed.
+
+        Args:
+            update_fields: Optional list/set of field names being updated. If provided,
+                          only these fields will be checked for changes.
+
+        Returns:
+            bool: True if any non-timestamp field has changed, False otherwise.
+        """
+        if not self.pk:
+            return True
+
+        model_cls = self.__class__
+        try:
+            old_instance = model_cls.objects.get(pk=self.pk)
+        except model_cls.DoesNotExist:
+            return True
+
+        # If update_fields is specified, only check those fields (minus excluded ones)
+        if update_fields is not None:
+            fields_we_can_check = set(update_fields)
+        else:
+            fields_we_can_check = [field.name for field in self._meta.get_fields() if field.concrete and hasattr(field, "name")]
```

Comment:
if you only care about concrete fields there is `<model>._meta.concrete_fields`

```suggestion
            fields_we_can_check = [field.name for field in self._meta.concrete_fields if hasattr(field, "name")]
```